include/networkit/graph/TopologicalSort.hpp

```diff
@@ -47,6 +58,10 @@ class TopologicalSort final : public Algorithm {
 
     const Graph *G;
 
+    std::optional<std::unordered_map<node, node>> computedNodeIdMap;
+
+    std::unordered_map<node, node> *nodeIdMap;
```

Comment:
I suggest using `const std::unordered_map<node, node> &nodeIdMap;` and change access in rest of the code accordingly.

Reply:
Can't do, for multiple reasons. First, C++ doesn't allow references as class members. Second, I can't have `const` and also use indexing by `[]`. Third, it is perfectly valid for `nodeIdMap` to be `null`.

Reply:
The part about C++ is not true. Take a look for example at `centrality/Centrality.hpp`. There the graph `const Graph& G` is a const reference member of the class.

Reply:
My bad, I mixed this up with a limitation in Microsoft's C++/CLI. Then I can indeed make this happen.